plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/internal/refresh/diagram/DDiagramCanonicalSynchronizer.java

```diff
@@ -236,22 +241,51 @@ private void manageRegions() {
+    private Optional<View> getSemanticAncestor(View view) {
+        return Stream.<EObject> iterate(view, child -> child.eContainer() instanceof View, EObject::eContainer) //
+                .map(View.class::cast) //
+                .filter(candidateAncestor -> candidateAncestor.getElement() != null) //
+                .findFirst();
+    }
+
+    private void updateLayoutOfMovedView(Map<View, View> movedViews, Map.Entry<View, View> movedView) {
+        View oldView = movedView.getKey();
+        View newView = movedView.getValue();
+        
+        Optional<View> newAncestor = getSemanticAncestor(oldView).map(movedViews::get);
+
+        if (oldView instanceof Node oldNode && newView instanceof Node newNode) {
+            LayoutConstraint oldLayoutConstraint = oldNode.getLayoutConstraint();
+            LayoutConstraint newLayoutConstraint = newNode.getLayoutConstraint();
+            if (oldLayoutConstraint instanceof Bounds oldBound && newLayoutConstraint instanceof Bounds newBound) {
+                if (newAncestor.isPresent()) {
+                    updateLocationConstraint(newBound, new org.eclipse.draw2d.geometry.Point(oldBound.getX(), oldBound.getY()));
+                }
+                updateSizeConstraint(newBound, new Dimension(oldBound.getWidth(), oldBound.getHeight()), newNode.getElement());
+                newView.eAdapters().remove(SiriusLayoutDataManager.INSTANCE.getAdapterMarker());
             }
-        });
+        }
+    }
+
+    private void manageCreatedViewsLayout(Set<View> createdViews, Map<View, View> movedViews) {
+        movedViews.entrySet().stream() //
+                .filter(moved -> this.createdViewIsMarkedAsToLayout(moved.getValue())) //
+                .forEach(movedView -> this.updateLayoutOfMovedView(movedViews, movedView));
+
+        // get view to layout "normally"
+        Set<View> filteredCreatedViewsToLayout = createdViews.stream() //
+                .filter(this::createdViewIsMarkedAsToLayout) //
+                .collect(Collectors.toSet());
+
+        // get view to layout "border node"
+        Set<View> filteredCreatedViewsToBorderNodeLayout = createdViews.stream() //
+                .filter(view -> view.eAdapters().contains(SiriusLayoutDataManager.INSTANCE.getBorderNodeMarker())) //
+                .collect(Collectors.toUnmodifiableSet());
 
         // get view to center layout
-        Set<View> filteredCreatedViewsWithCenterLayout = Sets.filter(createdViews, new Predicate<View>() {
-            @Override
-            public boolean apply(View input) {
-                return input.eAdapters().contains(SiriusLayoutDataManager.INSTANCE.getCenterAdapterMarker());
-            }
-        });
+        Set<View> filteredCreatedViewsWithCenterLayout = new LinkedHashSet<>(createdViews.stream() //
+                .filter(this::createdViewIsMarkedAsToCenterLayout) //
+                .collect(Collectors.toList()));
```

Comment:
toList() instead of collect(.. works fine

Reply:
**Warning** : If refresh is manual, when drag&dropping in the explorer the diagram is not refreshed at all
It does not seem to be a regression

Reply:
done